systemvm/patches/debian/config/opt/cloud/bin/cs/CsConfig.py

```diff
@@ -58,25 +58,35 @@ def get_level(self):
         return self.__LOG_LEVEL
 
     def is_vpc(self):
-        return self.cl.get_type() == "vpcrouter"
+        return self.cl.get_type() == 'vpcrouter'
 
     def is_router(self):
-        return self.cl.get_type() == "router"
+        return self.cl.get_type() == 'router'
+
+    def is_dhcp(self):
+        return self.cl.get_type() == 'dhcpsrvr'
+
+    def has_dns(self):
+        return not self.use_extdns()
+
+    def has_metadata(self):
+        return any((self.is_vpc(), self.is_router(), self.is_dhcp()))
 
     def get_domain(self):
         return self.cl.get_domain()
 
+    def use_extdns(self):
+        return self.cmdline().idata().get('useextdns', 'false') == 'true'
+
     def get_dns(self):
+        conf = self.cmdline().idata()
         dns = []
-        if not self.cl.get_use_ext_dns():
-            if not self.is_vpc() and self.cl.is_redundant():
```

Comment:
@DaanHoogland  As I can see, the CsGuestNetwork means the vpc tiers. 
For isolated networks/shared networks, the dns is set in CsConfig.get_dns().
So, I think the change in CsConfig.get_dns() is wrong.